lib/uktest/include/uk/test.h

```diff
@@ -0,0 +1,595 @@
+
+#ifdef CONFIG_LIBUKTEST_LOG_TESTS
+		/* Output the location where the expectation failed. */
+		uk_test_printf("  in " __STR_BASENAME__ ":"
+			       STRINGIFY(__LINE__) "\n");
+#endif /* CONFIG_LIBUKTEST_LOG_TESTS */
+
+#if CONFIG_LIBUKTEST_FAILFAST
+		UK_CRASH("Crashing on first failure!");
+#endif
+	}
+}
+
+/**
+ * Assert a boolean condition.
+ *
+ * @param esac
+ *   The test case under consideration.
+ * @param cond
+ *   The boolean condition of the assertion.
+ */
+#define UK_TEST_ASSERT(esac, cond) \
+	_uk_test_do_assert(esac, cond, STRINGIFY(cond))
+
+/**
+ * Assert a boolean condition with formatted string.
+ *
+ * @param esac
+ *   The test case under consideration.
+ * @param cond
+ *   The boolean condition to be evaluated for this assertion.
+ * @param fmt
+ *   Formatted string to print as help text to explain the assertion.
+ */
+#define UK_TEST_ASSERTF(esac, cond, fmt, ...) \
+	_uk_test_do_assert(esac, cond, fmt, ##__VA_ARGS__)
+
+/**
+ * Expect a condition to be true.
+ *
+ * @param esac
+ *   The test case under consideration.
+ * @param cond
+ *   The boolean condition which will render at runtime.
+ */
+#define UK_TEST_EXPECT(esac, cond)					\
+	UK_TEST_ASSERTF(esac,						\
+		cond,							\
+		"expected `" STRINGIFY(cond) "` to be true"		\
+	)
+
+/**
+ * Expect an expression to be NULL.
+ *
+ * @param esac
+ *   The test case under consideration.
+ * @param exp
+ *   The expression under consideration.s
+ */
+#define UK_TEST_EXPECT_NULL(esac, exp)					\
+	UK_TEST_ASSERTF(esac,						\
+		exp == NULL,						\
+		"expected `" STRINGIFY(exp) "` to be NULL"		\
+	)
+
+/**
+ * Expect an expression to not be NULL.
+ *
+ * @param esac
+ *   The test case under consideration.
+ * @param exp
+ *   The expression under consideration.s
+ */
+#define UK_TEST_EXPECT_NOT_NULL(esac, exp)				\
+	UK_TEST_ASSERTF(esac,						\
+		exp != NULL,						\
+		"expected `" STRINGIFY(exp) "` to not be NULL"		\
+	)
+
+/**
+ * Expect an expression to be evaluate to zero.
+ *
+ * @param esac
+ *   The test case under consideration.
+ * @param exp
+ *   The expression under consideration.s
+ */
+#define UK_TEST_EXPECT_ZERO(esac, exp)					\
+	UK_TEST_ASSERTF(esac,						\
+		exp == 0,						\
+		"expected `" STRINGIFY(exp) "` to be zero"		\
+	)
+
+/**
+ * Expect an expression to not evaluate to zero.
+ *
+ * @param esac
+ *   The test case under consideration.
+ * @param exp
+ *   The expression under consideration.s
+ */
+#define UK_TEST_EXPECT_NOT_ZERO(esac, exp)				\
+	UK_TEST_ASSERTF(esac,						\
+		exp != 0,						\
+		"expected `" STRINGIFY(exp) "` to not be zero"		\
+	)
+
+/**
+ * Expect two pointers to be equal to each other.
+ *
+ * @param esac
+ *   The test case under consideration.
+ * @param a
+ *   The left-hand operand.
+ * @param b
+ *   The right-hand operand.
+ */
+#define UK_TEST_EXPECT_PTR_EQ(esac, a, b)				\
+	UK_TEST_ASSERTF(esac,						\
+		(a) == (b),						\
+		"expected `" #a "` and `" #b "` to be %p "		\
+		"but was %p",						\
```

Comment:
Perhaps should clarify that b has a different value

Reply:
What do you mean?

Reply:
Ok, maybe it is the way I read that message. For me it is: "a and b should be X, but it was Y." and then I ask myself, what "it" refers to, a or b.